assets/js/payment-method-extensions/payment-methods/stripe/credit-card/index.js

```diff
@@ -36,11 +36,11 @@ const StripeComponent = ( props ) => {
 const StripeLabel = ( props ) => {
 	const { PaymentMethodLabel } = props.components;
 
-	return (
-		<PaymentMethodLabel
-			text={ __( 'Credit / Debit Card', 'woo-gutenberg-products-block' ) }
-		/>
-	);
+	const labelText = getStripeServerData().title
+		? getStripeServerData().title
+		: __( 'Credit / Debit Card', 'woo-gutenberg-products-block' );
```

Comment:
Maybe we can skip the `?` check and use directly `getStripeServerData().title` in this place since we are already testing this on the API side?

Reply:
This gives us an extra check on client, we shouldn't rely on the server default IMO - just more robust to have a late check. Also the behaviour for empty string (`title === ''`) might be different in JS vs PHP.